src/discounts/views/SaleDetails/handlers.ts

```diff
@@ -30,22 +30,26 @@ export function createUpdateHandler(
     variables: SaleChannelListingUpdateVariables;
   }) => Promise<MutationFetchResult<SaleChannelListingUpdate>>
 ) {
-  return (formData: SaleDetailsPageFormData) => {
+  return async (formData: SaleDetailsPageFormData) => {
     const { id } = sale;
-    updateSale({
-      id,
-      input: {
-        endDate: formData.hasEndDate
-          ? joinDateTime(formData.endDate, formData.endTime)
-          : null,
-        name: formData.name,
-        startDate: joinDateTime(formData.startDate, formData.startTime),
-        type: discountValueTypeEnum(formData.type)
-      }
-    });
+    const errors = await Promise.all([
+      updateSale({
+        id,
+        input: {
+          endDate: formData.hasEndDate
+            ? joinDateTime(formData.endDate, formData.endTime)
+            : null,
+          name: formData.name,
+          startDate: joinDateTime(formData.startDate, formData.startTime),
+          type: discountValueTypeEnum(formData.type)
+        }
+      }).then(({ data }) => data?.saleUpdate.errors ?? []),
 
-    updateChannels({
-      variables: getSaleChannelsVariables(id, formData, saleChannelsChoices)
-    });
+      updateChannels({
+        variables: getSaleChannelsVariables(id, formData, saleChannelsChoices)
+      }).then(({ data }) => data?.saleChannelListingUpdate.errors ?? [])
```

Comment:
The thought of having `then` is itching me but I can't really request changes here

Reply:
True. I could use IIFE with awaits but I think this looks cleaner.
```
      async () => {
        const { data } = await updateChannels({
          variables: getSaleChannelsVariables(id, formData, saleChannelsChoices)
        });
        return data?.saleChannelListingUpdate.errors ?? [];
      }
```